app/src/test/kotlin/com/intive/picover/common/viewmodel/state/ViewModelStateTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.intive.picover.common.viewmodel.state
+
+import io.kotest.assertions.assertSoftly
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.inspectors.forAll
+import io.kotest.matchers.shouldBe
+import io.mockk.mockk
+
+internal class ViewModelStateTest : ShouldSpec(
+	{
+		val data: Any = mockk()
+
+		should("SHOULD isLoading AND isError AND isLoaded AND data return proper values based on view model state") {
+			listOf(
+				ViewModelStateParams(
+					state = ViewModelState.Loading,
+					isLoading = true,
+					isError = false,
+					isLoaded = false,
+					data = null,
+				),
+				ViewModelStateParams(
+					state = ViewModelState.Error,
+					isLoading = false,
+					isError = true,
+					isLoaded = false,
+					data = null,
+				),
+				ViewModelStateParams(
+					state = ViewModelState.Loaded(data),
+					isLoading = false,
+					isError = true,
+					isLoaded = true,
+					data = data,
+				),
+			).forAll { (viewModelState, isLoading, isError, isLoaded, data) ->
```

Comment:
Kotest gives us `forAll` collection inspectors, which can be used in the tests. 

Output from this is very nice:

2 elements passed but expected 3

The following elements passed:
ViewModelStateParams(state=com.intive.picover.common.viewmodel.state.ViewModelState$Loading@682a52b8, isLoading=true, isError=false, isLoaded=false, data=null)
ViewModelStateParams(state=com.intive.picover.common.viewmodel.state.ViewModelState$Error@80ecae9, isLoading=false, isError=true, isLoaded=false, data=null)

The following elements failed:
ViewModelStateParams(state=Loaded(data=Any(#1)), isLoading=false, isError=true, isLoaded=true, data=Any(#1)) => The following assertion for Loaded(data=Any(#1)) failed:
expected:<true> but was:<false>
Expected :true
Actual :false

So i think we won't need the dynamic test name for every variant now.
There are also `withData` tests but in my opinion these are quite nice and looks like a candidate for our schema.